pkg/purge/purge.go

```diff
@@ -0,0 +1,221 @@
+
+	for _, obj := range objs {
+		gvk := obj.GroupVersionKind()
+		mapping, err := c.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			return fmt.Errorf("could not get restmapping for %v %v: %w", obj.GetKind(), obj.GetName(), err)
+		}
+
+		bar.Describe(fmt.Sprintf("Deleting %v (%v)", obj.GetName(), obj.GetKind()))
+		err = c.client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
+			Delete(ctx, obj.GetName(), metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("could not delete %v %v: %w", obj.GetKind(), obj.GetName(), err)
+		} else if errors.IsNotFound(err) {
+			fmt.Printf("Resource %v %v not found, skipping.\n", obj.GetKind(), obj.GetName())
+		}
+
+		_ = bar.Add(1)
+	}
+
+	return nil
+}
+
+func (c *PurgeClient) preprocessManifests(
+	ctx context.Context,
+	objs []unstructured.Unstructured,
+	options *PurgeOptions,
+) error {
+	var compositeErr error
+	for _, obj := range objs {
+		gvk := obj.GroupVersionKind()
+		mapping, err := c.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			return err
+		}
+		existing, err := c.client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
+			Get(ctx, obj.GetName(), metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		} else if err == nil {
+			if _, ok := existing.GetAnnotations()["kubectl.kubernetes.io/last-applied-configuration"]; ok {
+				multierr.AppendInto(&compositeErr,
+					fmt.Errorf("%v %v has been modified with kubectl", obj.GetKind(), obj.GetName()))
+			}
+		}
+
+		if obj.GetKind() == "Namespace" && obj.GetName() == "glasskube-system" {
+			nsAnnotations := obj.GetAnnotations()
+			if nsAnnotations == nil {
+				nsAnnotations = make(map[string]string, 1)
+			}
+			if existing != nil {
+				existingAnnotations := existing.GetAnnotations()
+				nsAnnotations[annotations.TelemetryEnabledAnnotation] = existingAnnotations[annotations.TelemetryEnabledAnnotation]
+				nsAnnotations[annotations.TelemetryIdAnnotation] = existingAnnotations[annotations.TelemetryIdAnnotation]
+			}
+			annotations.UpdateTelemetryAnnotations(nsAnnotations, options.DisableTelemetry)
+			obj.SetAnnotations(nsAnnotations)
+			options.DisableTelemetry = !annotations.IsTelemetryEnabled(nsAnnotations)
+		}
+	}
+
+	if compositeErr != nil {
+		compositeErr = fmt.Errorf("unsupported installation: %w", compositeErr)
+	}
+	return compositeErr
+}
+
+func (c *PurgeClient) handleTelemetry(disabled bool, elapsed time.Duration) {
+	if !disabled {
+		statusMessage("Telemetry is enabled for this cluster – "+
+			"Run \"glasskube telemetry status\" for more info.", true)
+		telemetry.BootstrapSuccess(elapsed)
+	}
+}
+
+func statusMessage(input string, success bool) {
+	if success {
+		green := color.New(color.FgGreen).SprintFunc()
+		fmt.Println(green("* " + input))
+	} else {
+		red := color.New(color.FgRed).SprintFunc()
+		fmt.Println(red("* " + input))
+	}
+}
```

Comment:
I understand that you just copied this part from the `bootstrap` package, but we pivoted to using the `statuswriter` package for this, as well as handling the progress bar. Please consider migrating to that package instead. You can take  `pkg/install` as reference.